PKPD-30 collapsible sidebar

wniestroj opened the pull request:
Added collapsible sideber, added animations to main content and simulations sidebar

eatyourgreens commented on frontend-v2/src/features/main/Sidebar.tsx:
```diff
@@ -214,7 +218,34 @@ export default function Sidebar() {
-  const drawer = (
+  const [isExpanded, setIsExpanded] = useState(true);
+  const onCollapse = () => {
+    const element = document.getElementById("main-content");
+    element?.classList.remove("on-expand");
+    element?.classList.add("on-collapse");
+
+    const simulationsNav = document.getElementById("simulations-content");
+    if (simulationsNav) {
+      simulationsNav?.classList.remove("on-expand");
+      simulationsNav?.classList.add("on-collapse");
+    }
```
This logic could maybe move into a function outside of the component, to make it clear that the DOM is being modified outside of React.

Also, the DOM changes here won't be tracked by React's virtual DOM, so I'm not 100% sure if they'll persist across re-renders.

eatyourgreens commented on frontend-v2/src/features/main/Sidebar.tsx:
```diff
@@ -214,7 +218,34 @@ export default function Sidebar() {
+  };
+
+  const onExpand = () => {
+    const element = document.getElementById("main-content");
+    element?.classList.remove("on-collapse");
+    element?.classList.add("on-expand");
+
+    const simulationsNav = document.getElementById("simulations-content");
+    if (simulationsNav) {
+      simulationsNav?.classList.remove("on-collapse");
+      simulationsNav?.classList.add("on-expand");
+    }
```
Same comments here about modifying the DOM outside of the React virtual DOM.

eatyourgreens commented on frontend-v2/src/features/main/Sidebar.tsx:
```diff
@@ -368,7 +487,12 @@ export default function Sidebar() {
+          },
+          flexShrink: { sm: 0 },
+        }}
         aria-label="mailbox folders"
```
Not related to this PR, but I just noticed that these ARIA labels are wrong.

eatyourgreens commented on frontend-v2/src/features/main/Sidebar.tsx:
```diff
@@ -407,23 +533,26 @@ export default function Sidebar() {
-            backgroundColor: '#FBFBFA',
-            borderRight: '1px solid #DBD6D1'
           }}
           aria-label="mailbox folders"
```
Also not related to this PR, but the wrong ARIA label here too.

eatyourgreens commented on frontend-v2/src/features/main/Sidebar.tsx:
```diff
@@ -407,23 +533,26 @@ export default function Sidebar() {
       )}
       <Box
         component="main"
+        id="main-content"
```
I'd be inclined to set the expanded/collapsed className here, rather than modify the DOM directly.

eatyourgreens reviewed (approved):
Changes LGTM. I'm not 100% sure about modifying the DOM directly, and whether that will work with React's virtual DOM.

eatyourgreens commented on frontend-v2/src/features/main/Sidebar.tsx:
```diff
@@ -368,8 +464,13 @@ export default function Sidebar() {
+          },
+          flexShrink: { sm: 0 },
+        }}
+        aria-label="navigation"
```
The ARIA role of a HTML `<nav>` is 'navigation'. Screen-readers announce the role and label so this would be something like 'navigation navigation'. I'd recommend using `aria-label` to give this `<nav>` a more specific label.
```suggestion
        aria-label="project navigation"
```

eatyourgreens commented:
@wniestroj Changes look good to me. I've one small comment about avoiding generic labels for landmark elements like `<nav>`. Here's a screen recording showing the landmarks as they are announced by VoiceOver with Chrome. Each landmark area of the app is labelled with both its role and accessible name (`aria-label`.)

By the way, when I collapse the sidebar for Simulations, there's a lot of whitespace on the right. The plots don't fill the available width.

https://github.com/user-attachments/assets/08fee297-ba14-495e-beba-1d79f3f2bd2d

wniestroj commented:
@eatyourgreens I wasn't able to listen to your recording on that point about screen readers, however I have tested it myself and I think I understood your point. At first I have applied your proposed change to aria-label to 'project navigation' however, that also caused word 'navigation' be repeated (project navigation navigation), therefore I've decided to change the aria-label to 'main' so that this element is voiced as 'main navigation'. Feel free to propose other labels and let me know if you would have any additional points regarding the screen readers

Regarding first video of a lot of white space on simulations page - this recorded view seems odd, I'm assuming you might've been resizing the window, which didn't update the plot widths. I have added two things:
1. event handler for plots to get resized, whenever the window is resized, so the plot widths should be updated when user resizes the window, or changes resolution automatically
2. Added 'saved' width by collapsing the navigation to the plot widths, so those should get larger by the same amount that is freed up by collapsing the navigation. Works both ways of course.

Regarding the second video, I have tested it throughly before creating a merge request and never had such issue. I haven't tested the code after your merge of ui-enchancements to the branch, as somehow automatically I have merged the ui-enchancements branch to this branch and tested it again - and I haven't been able to replicate this issue as well. I've pushed both mentioned changes, as well as merge of the ui-ench... branch. Could you check if this issue is still occurring? And if so, could you provide a path to recreating this issue?

eatyourgreens commented:
Re. the simulations view, I hadn't resized the window but here's a recording where I enlarge the window from very narrow to very wide. The plot starts off overflowing the viewport width, then shrinks (with white space to the right), then switches to a two column grid.


https://github.com/user-attachments/assets/76afb91d-9c0b-477b-bc04-b069cfa00a0b

wniestroj commented:
I have refactored the way the widths of the plots, dropping breakpoints solution and calculating width of plots in dependence to parent width (not pushed yet), however, I'm unable to find a solution to those animations not being triggered some of the times. I'll get back to it first thing tomorrow, but I'm afraid I might need to drop the animations entirely if I'm unable to make it work

wniestroj commented:
@eatyourgreens I have updated how plot widths are calcualted, so that they should always take up all available space. Note: I have added a half a second debounce for performance issues, so that it does not enter an infinite loop of resizing, therefore continuous resizing will not trigger the resizing of plots, only .5s after the resizing the widths will get updated.

Regarding your issue of animation not always triggering, after alignment with MIchael, Martin and Stefanie, we have decided not to pursue a fix for that, as it seems to be working fine on Chrome, and within our organization we are forced to use Chrome (we are unable to even download Firefox).

I have also raised concerns about some performance issues of Navigation animation - I'm not sure is those are caused by my machine being constantly overloaded, or if those issues are caused by something else. We have decided to go ahead and push my changes to the environment and in case those issues are still present after the deployment, we will either get rid of the animations as a whole, or remove animations from the sidebar

eatyourgreens commented:
Sounds good to me. One improvement that I would suggest: remember the expanded/contracted state of the sidebar across sessions (maybe in a cookie or local storage.) At the moment, the sidebar is always expanded when I reload the page or open a new tab.
